RHEL6_7/networking/dovecot/check

```diff
@@ -69,13 +78,48 @@ file manually on the new system.
 PREF=$POSTUPGRADE_DIR/dovecot
 mkdir -p $PREF
 sed '2,/^#!\//d' $0 >$PREF/dovecot_postupgrade.sh
-if ! [[ -z "$uid_val" ]] && [ "$uid_val" -lt 1000 ];then
-  export uid_val ; perl -pi -e '$_ .= qq(sed -i -e "s/first_valid_uid = 1000/first_valid_uid = $ENV{"uid_val"}/g" /etc/dovecot/conf.d/10-mail.conf\n) if  /restorecon -R $CONFIG_FILE/' "$PREF"/dovecot_postupgrade.sh
-fi
 chmod +x $PREF/dovecot_postupgrade.sh
 
 rm -f $TMPF1 $TMPF2
-exit $RESULT_FIXED
+cd /etc/dovecot/
+
+#Get the the config, that actually provides the first_valid_uid directive
+if [[ "${included_files[@]}" != "" ]];then
+    printf '%s\n' ${included_files[@]} | while IFS= read -r included_file 
+    do
+        if [ -e "$included_file" ];then
+           if grep -q "^first_valid_uid" "$included_file" ;then
+               included_file="$included_file"
+           echo "$included_file" > "$loaded_conf"
+           fi
+
+        fi
+    done
+else
+    if grep -q "^first_valid_uid" "/etc/dovecot/dovecot.conf" ;then
+        echo "/etc/dovecot/dovecot.conf" > "$loaded_conf"
+    fi
+fi
+last_loaded=$(cat "$loaded_conf")
+last_loaded="$VALUE_TMP_PREUPGRADE/dirtyconf/$CONFIG_FILE/$last_loaded"
+export uid_val 
+export last_loaded
+
+if [[ "$explicit_uid_val" == "" ]];then
+    #Change the explcitly stated default value on the target system to the default value on the source system
+    perl -pi -e '$_ .= qq(first_valid_uid = $ENV{uid_val}\n) if /^#first_valid_uid = 500/' "$VALUE_TMP_PREUPGRADE/dirtyconf/$CONFIG_FILE/conf.d/10-mail.conf"
```

Comment:
Issues:
 - what if the line `first_valid_uid = 500` doesn't exist inside the file?
 - what if the file does not exist?